assets/prototype/domain/shared/predicates/datetimes/validInfiniteReglimit.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * External dependencies
+ */
+import { is } from 'ramda';
+
+/**
+ * Internal dependencies
+ */
+import { Datetime } from '../../../eventEditor/data/types';
+
+/**
+ * @param {Object} date event date object
+ * @return {boolean} true if regLimit property is valid
+ */
+const validReglimit = (date: Datetime) => {
+	return is(String, date.regLimit) || is(Number, date.regLimit);
+};
+
+/**
+ * @param {Object} date event object
+ * @return {boolean} true if regLimit property is valid and unlimited
+ */
+const validInfiniteReglimit = (date: Datetime) => {
+	return validReglimit(date) && (date.regLimit === 'INF' || date.regLimit === Infinity);
```

Comment:
@manzoorwanijk what is the type of `regLimit` on the schema ?

Reply:
See [this comment](https://github.com/eventespresso/event-espresso-core/pull/2087#discussion_r361135896).